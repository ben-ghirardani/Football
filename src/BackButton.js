import React, { Component } from 'react';
import Button from './StyledComponents/Button';

export default class BackButton extends Component {

    constructor() {
        super();
        this.onClick = this.onClick.bind(this);
    }

    onClick() {
        console.log('clickety click')
        if(this.props.display === "teamSelected") {
            this.props.switchViewComponent("table")
        } else if (this.props.display === "match") {
            this.props.switchViewComponent("teamSelected")
        }
        else return
    }

    render() {
        return(
            <Button onClick={this.onClick}>
                Back
            </Button>
        )
    }

}