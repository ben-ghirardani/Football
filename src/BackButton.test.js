import React, { Component } from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import BackButton from './BackButton';

// onClick in <BackButton> uses functions from App passed as props so test them there
describe('BackButton', () => {
    it('should render', () => {
        const component = shallow(<BackButton/>);
        
        expect(component).toMatchSnapshot();
      });
  });