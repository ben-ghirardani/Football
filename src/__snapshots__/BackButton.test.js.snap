// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Back",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Back",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Back",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Back",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
